main.cpp

```diff
@@ -131,41 +202,190 @@ std::pair<JSONObject, size_t> parse(std::string_view json) {
         }
         return {JSONObject{std::move(res)}, i};
     } else if (json[0] == '{') {
+        //for dict
         std::unordered_map<std::string, JSONObject> res;
         size_t i;
         for (i = 1; i < json.size();) {
             if (json[i] == '}') {
                 i += 1;
                 break;
             }
-            auto [keyobj, keyeaten] = parse(json.substr(i));
-            if (keyeaten == 0) {
-                i = 0;
-                break;
-            }
-            i += keyeaten;
-            if (!std::holds_alternative<std::string>(keyobj.inner)) {
-                i = 0;
-                break;
-            }
-            if (json[i] == ':') {
-                i += 1;
+            std::string key;
+            {//for key
+                std::regex elem_re(R"([ \n\r\t\v\f\0]*?[\'\"]{0,1}[a-zA-Z]+[\'\"]{0,1}[ \n\r\t\v\f\0]*?:)");
+                std::cmatch match;
+                std::regex_search(json.data() + i, json.data() + json.size(), match, elem_re);
+                key = match.str();
+                i += key.size();
+                key = key.substr(0, key.size() - 1);
```

Comment:
```suggestion
key.pop_back();
```
substr causes memory allocation.